src/plugins/mer/mervirtualmachinesettingswidget.cpp

```diff
@@ -57,6 +57,24 @@ MerVirtualMachineSettingsWidget::~MerVirtualMachineSettingsWidget()
     delete ui;
 }
 
+void MerVirtualMachineSettingsWidget::setVmFeatures(VirtualMachine::Features features)
+{
+    m_features = features;
+    QString stopVmText = tr("Stop the virtual machine to unlock this field for editing.");
+    if (features & VirtualMachine::LimitMemorySize)
+        setToolTip(ui->memoryInfoLabel, stopVmText);
+    else
+        setToolTip(ui->memoryInfoLabel, tr("Virtual Machine does not allow changing memory size"));
+    if (features & VirtualMachine::LimitCpuCount)
+        setToolTip(ui->cpuInfoLabel, stopVmText);
+    else
+        setToolTip(ui->cpuInfoLabel, tr("Virtual Machine does not allow changing cpu count"));
+    if (features & (VirtualMachine::GrowStorageSize | VirtualMachine::ShrinkStorageSize))
```

Comment:
We should also update the min/max limits based on that or at least add an assertion that the ShrinkStorageSize feature is not enabled.

Reply:
We are getting quite far from the bug that I'm fixing here...